modules/iam-circleci-oidc-role/variables.tf

```diff
@@ -0,0 +1,92 @@
+variable "create" {
+  description = "Controls if resources should be created (affects all resources)"
+  type        = bool
+  default     = true
+  nullable    = false
+}
+
+variable "tags" {
+  description = "A map of tags to add to the resources created"
+  type        = map(any)
+  default     = {}
+  nullable    = false
+}
+
+################################################################################
+# CircelCI OIDC Role
+################################################################################
+
+variable "name" {
+  description = "Name of IAM role"
+  type        = string
+  default     = null
+}
+
+variable "path" {
+  description = "Path of IAM role"
+  type        = string
+  default     = "/"
+  nullable    = false
+}
+
+variable "permissions_boundary_arn" {
+  description = "Permissions boundary ARN to use for IAM role"
+  type        = string
+  default     = null
+}
+
+variable "description" {
+  description = "IAM Role description"
+  type        = string
+  default     = null
+}
+
+variable "name_prefix" {
+  description = "IAM role name prefix"
+  type        = string
+  default     = null
+}
+
+variable "policies" {
+  description = "Policies to attach to the IAM role in `{'static_name' = 'policy_arn'}` format"
+  type        = map(string)
+  default     = {}
+  nullable    = false
```

Comment:
Nullables are not needed in most places in the module. At least, I think so. We don't have them in pretty much any modules.

@bryantbiggs What do you think?

Reply:
IMHO, it is very useful to set `nullable = false` anywhere you do not expect to get `null`s so you do not need to handle situations when `null` is set and there is an expected default. Also it is useful in sub modules for handling defaults by setting null in the input to make that default of the variable take place.

We do it for all our modules :)

But, your call, I can remove if you think it is better for the community.

Reply:
I don't see the value in using `nullable` - we haven't seen a use case yet where the resolution was to use `nullable`

Reply:
Ok